src/Workspaces/Core/Portable/EmbeddedLanguages/RegularExpressions/LanguageServices/RegexDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,93 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Threading;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.EmbeddedLanguages.LanguageServices;
+using Microsoft.CodeAnalysis.Options;
+
+namespace Microsoft.CodeAnalysis.EmbeddedLanguages.RegularExpressions.LanguageServices
+{
+    /// <summary>
+    /// Analyzer that reports diagnostics in strings that we know are regex text.
+    /// </summary>
+    internal class RegexDiagnosticAnalyzer : IEmbeddedDiagnosticAnalyzer
+    {
+        public const string DiagnosticId = "RE001";
```

Comment:
>RE001 [](start = 44, length = 5)

Why not continue the existing `IDEnnn` sequence? (in other words, why use a different prefix?)

Reply:
These seemed different enough to warrant having their own potential IDs that were not combined with those other features.

Reply:
I don't have a strong opinion on this. I thought it's strange because I expect it will be the only one with that prefix. I'll let IDE team provide guidance.

Reply:
It won't necessarily be the only one though.  For example, you could imagine the IDE supplying other diagnostics and fixes on regexes.  For example, an analyzer that said ```(a|b|a)`` is redundant.  Or ```a**``` isn't necessary.  It was something on my list i wanted to do, but i timeboxed myself out of it.